### PR TITLE
Refactor IAP

### PR DESCRIPTION
**Pull request by blackzlq**
Refactor IAP module

Tested-by: zlq on AP cluster and standard

### CODE REVIEW

File: applications/jupyter/main.tf [hunk 1]
```diff
@@ -77,6 +77,16 @@ module "namespace" {
   create_namespace = true
 }
 
+# IAP Section: Enabled the IAP service
+resource "google_project_service" "project_service" {
```
Comment (umeshkumhar) [inline applications/jupyter/main.tf:81]: can this be moved into iap module

Comment (blackzlq) [reply applications/jupyter/main.tf:81]: Discussed offline, we will do another test later. From my option this is project level, should be enabled once. and Umesh thinks it is good for code maintain and generality of IAP module, also the terraform will auto correct it.

File: modules/iap/iap.tf [hunk 2]
```diff
@@ -12,152 +12,98 @@
...
   provider     = google-beta
   project      = var.project_id
-  name         = "frontend-address-${random_string.random.result}"
+  name         = "${var.app_name}-address-${random_string.random.result}"
   address_type = "EXTERNAL"
   ip_version   = "IPV4"
 }
 
 # Helm Chart IAP
 resource "helm_release" "iap_frontend" {
```
Comment (umeshkumhar) [inline modules/iap/iap.tf:53]: rename iap_frontend to iap

**Review by umeshkumhar — APPROVED** [review]

File: modules/iap/iap.tf (hunk 2 above)
Comment (blackzlq) [reply modules/iap/iap.tf]: Good catch!

**Pull request merged**